web/src/routes/+layout.svelte

```diff
@@ -104,6 +107,32 @@
   };
 
   $effect(() => void handleRelease($release));
+
+  serverRestarting.subscribe((isRestarting) => {
+    if (!isRestarting) {
+      return;
+    }
+
+    if (isRestarting.isMaintenanceMode !== page.url.pathname.startsWith(AppRoute.MAINTENANCE)) {
```

Comment:
You have this maintenance route check three times now, let's put it into a utility method like `isMaintenanceRoute` or something.